src/code/z_demo.c

```diff
@@ -333,202 +367,193 @@ void Cutscene_Command_Misc(PlayState* play, CutsceneContext* csCtx, CsCmdBase* c
                 break;
 
-            case 8:
-                if (seqId != NA_BGM_DISABLED) {
-                    Audio_PlaySceneSequence(seqId, dayMinusOne);
+            case CS_MOD_SEQ_RESTORE:
+                if (sSeqId != NA_BGM_DISABLED) {
+                    Audio_PlaySceneSequence(sSeqId, dayMinusOne);
                 }
                 break;
+
+            default:
+                break;
         }
     }
 }
 #else
-void func_800EADB0(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd);
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EADB0.s")
+void CutsceneCmd_ModifySequence(PlayState* play, CutsceneContext* csCtx, CsCmdModifySeq* cmd);
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/CutsceneCmd_ModifySequence.s")
 #endif
 
-// Command 0x12F: Fade Ambience sequence
-void Cutscene_Command_FadeAmbienceSequence(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
-    if (csCtx->frames == cmd->startFrame && csCtx->frames < cmd->endFrame) {
-        u8 fadeTimer = cmd->endFrame - cmd->startFrame;
+void CutsceneCmd_FadeOutAmbience(PlayState* play, CutsceneContext* csCtx, CsCmdFadeOutAmbience* cmd) {
+    if ((csCtx->curFrame == cmd->startFrame) && (csCtx->curFrame < cmd->endFrame)) {
+        u8 fadeOutDuration = cmd->endFrame - cmd->startFrame;
 
-        Audio_QueueSeqCmd((fadeTimer << 0x10) | 0x140000FF);
+        Audio_QueueSeqCmd((fadeOutDuration << 0x10) | 0x140000FF);
     }
 }
 
-// Command 0x190: Rumble
-void Cutscene_Command_Rumble(PlayState* play, CutsceneContext* csCtx, CsCmdRumble* cmd) {
+void CutsceneCmd_RumbleController(PlayState* play, CutsceneContext* csCtx, CsCmdRumble* cmd) {
     switch (cmd->type) {
-        case 1:
-            if (csCtx->frames == cmd->startFrame) {
+        case CS_RUMBLE_ONCE:
+            if (csCtx->curFrame == cmd->startFrame) {
                 Rumble_Request(0.0f, cmd->intensity, cmd->decayTimer, cmd->decayStep);
             }
             break;
 
-        case 2:
-            if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames)) {
-                if ((csCtx->frames == cmd->startFrame) || (play->state.frames % 64 == 0)) {
+        case CS_RUMBLE_REPEATED:
+            if ((csCtx->curFrame >= cmd->startFrame) && (csCtx->curFrame <= cmd->endFrame)) {
+                if ((csCtx->curFrame == cmd->startFrame) || (play->state.frames % 64 == 0)) {
                     Rumble_Request(0.0f, cmd->intensity, cmd->decayTimer, cmd->decayStep);
                 }
             }
             break;
+
+        default:
+            break;
     }
 }
 
-// Command 0x9B:
-void Cutscene_Command_FadeColorScreen(PlayState* play, CutsceneContext* csCtx, CsCmdFadeScreen* cmd) {
-    if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames)) {
+void CutsceneCmd_FadeColorScreen(PlayState* play, CutsceneContext* csCtx, CsCmdFadeScreen* cmd) {
+    if ((csCtx->curFrame >= cmd->startFrame) && (cmd->endFrame >= csCtx->curFrame)) {
         f32 alpha;
```

Comment:
this is interesting. this is just an infinitely better implemented version of oot's transition command. but oots still exists here for some reason?

it feels like this should probably get the name transition, but i guess it needs more context (which is used more? stuff like that)

Reply:
Oh yeah, they really are just similar and this one is a cleaner, general one. Both are used heavily: This general `FadeScreen` is used 127 times, while the legacy transition one is still used 154 times.

I could see this being `CS_TRANSITION`, the question is what to name the other one. Maybe `CS_TRANSITION_SPEC`? as in special? It's like limited cases of `TRANSITION`. Alternatively, could leave the other one as `CS_TRANSITION` and name this one `CS_TRANSITION_GENERAL`

Reply:
yeah if theyre used about the same amount then I dont really know. wonder if theyre used for similar purposes or what
itd be interesting to see if the new command is only used for colors that the oot one doesnt have. 
I can see a situation where they left the old one cause their tools are already hooked up to use it, but in other scenarios they use the new one if the color isnt already defined (as opposed to adding new colors to oots)

Reply:
I'm seeing about 80% of use cases of `FADE_SCREEN` using either black or white, but there are some different color choices in there

Reply:
general works, or maybe "custom color"
but also if nearly all uses are black and white, its weird to use the "custom" command for something that the non-custom one already defines lol

would be interested to see a comparison of the two commands. surely they look identical when the color provided is the same as one of the options from the oot command?
thats why I still wonder if usage shows that theres a difference, but idk what else youd use it for other than screen transitions

Reply:
Yeah, every `CutsceneCmd_Transition` command from Oot And MM can be made with this `CutsceneCmd_TransitionGeneral`/`CutsceneCmd_TransitionColored` command with only small exceptions.

- It can't capture the sfx made by `CS_TRANS_GRAY_FILL_IN`
- In OoT, values 12/13 `CS_TRANS_BLACK_FILL_OUT_TO_HALF`/`CS_TRANS_BLACK_FILL_IN_FROM_HALF` can't be captured by `CutsceneCmd_TransitionGeneral`
- In MM, values 12/13 `CS_TRANS_GRAY_TO_BLACK`/`CS_TRANS_BLACK_TO_GRAY` can't be captured by `CutsceneCmd_TransitionGeneral`